### DISCUSSION TITLE
Bug fix/Issue History

### DISCUSSION THREAD

**Discussion by DderrickLee** [ucsb-cs156-f24/team02-f24-10#66] [Announcements]
## 👋 Welcome!
  List issues/bugs you've encountered and how you fixed them. Keep it short. These should be bugs that troubled you for a decent amount of time.

**Comment by AjayLiu** [comment]
### About the Table Component issue:

If you're stuck, check out this video, because apparently you're supposed to comment some lines out until a later team assignment: https://www.youtube.com/watch?v=znNAWKBwhQs

**Comment by AjayLiu** [comment]
### If you have issues passing the formatting Git check

My goat Sam found a fix, make sure you're in `/frontend` directory and run
```
npm run format
```

**Reply by r0ckwav3** [reply]
I was also having issues with prettier not being installed for this step, but you can install it with `pnpm add --save-dev --save-exact prettier`. Then you need to make a `.prettierrc` file in your project root with `{}` in it and vscode should be able to run it.

**Comment by AjayLiu** [comment]
### If you are failing Stryker Mutation testing with this ```.toBeCalled()``` function

Change it to ```.toBeCalledWith()``` and try again, that fixed it for me

**Comment by AjayLiu** [comment]
### If you're working on Index Page and can't pass tests dealing with the delete button

We commented out some things in the ___Table.js in one of the earliest tickets and we have to uncomment them now. In particular, you have to introduce a ___Utils.js file to help with the deleting

**Reply by justinwu13** [reply]
this and also you need to uncomment out the test file for the table too, I forgot about that at first

**Comment by gargAneesh** [comment]
I'm having an issue with migrating the backend stuff with my postgres database - for some reason my database is not showing when i do \db in the postgres console thing. I checked my environment variables and they seem to be fine and I also re-made the database.. not sure why this is the case.

**Reply by justinwu13** [reply]
are you checking to see if your table shows up? the command for that should be `\dt`, I don't think we've ever used `\db`

**Comment by r0ckwav3** [comment]
### If you're getting Stryker failures on test-ids in the form component.
If you're getting stryker mutation failures corresponding to lines like `data-testid={testIdPrefix + "-orgCode"}` you can add these lines into one of the tests:
```
await screen.findByTestId(`${testId}-orgCode`);
    expectedIds.forEach((idText) => {
      const testid = screen.getByTestId(`${testId}-${idText}`);
      expect(testid).toBeInTheDocument();
    });
```

you also need to define `expectedIds`, which should be a list of the testids (without the dashes).

**Reply by r0ckwav3** [reply]
also change `orgCode` to any testid that will always exist (i.e. your id field)

**Comment by r0ckwav3** [comment]
### Fails in the edit page toast print check
I got a strange error where my edit page worked great when I tested it manually but gave the wrong result to MockToast in the testfile. The issue was that the simulated PUT endpoint needs to return the edited version of the object rather than the unedited one which the GET endpoint returns.

**Comment by r0ckwav3** [comment]
### WHEN STARTING TEAM03 MAKE SURE TO MERGE f24
instead of 
```
git remote add starter https://github.com/ucsb-cs156-s24/STARTER-team02
```
use
```
git remote add starter https://github.com/ucsb-cs156-f24/STARTER-team02
```
